low-liquidity-market-attack-monitor/src/agent.js

```diff
@@ -0,0 +1,165 @@
+const {
+  Finding, FindingSeverity, FindingType, ethers, getEthersBatchProvider,
+} = require('forta-agent');
+
+const CERC20_MINT_EVENT = 'event Mint(address minter, uint256 mintAmount, uint256 mintTokens)';
+const ERC20_TRANSFER_EVENT = 'event Transfer(address indexed from, address indexed to, uint256 amount)';
+
+const { getAbi } = require('./utils');
+
+// load any agent configuration parameters
+const config = require('../agent-config.json');
+
+// set up a variable to hold initialization data used in the handler
+const initializeData = {};
+
+// helper function to create cToken alerts
+function createMarketAttackAlert(
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation,
+  compTokenSymbol,
+  compTokenAddress,
+  mintAmount,
+  mintTokens,
+  maliciousAddress,
+  maliciousAmount,
+) {
+  const finding = Finding.fromObject({
+    name: `${protocolName} cToken Market Attack Event`,
+    description: `The address ${maliciousAddress} is potentially manipulating the cToken ${compTokenSymbol} market`,
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-MARKET-ATTACK-EVENT`,
+    type: FindingType.Suspicious,
+    severity: FindingSeverity.Info,
+    protocol: protocolName,
+    metadata: {
+      compTokenSymbol,
+      compTokenAddress,
+      mintAmount,
+      mintTokens,
+      maliciousAddress,
+      maliciousAmount,
+    },
+  });
+  return finding;
+}
+
+async function getCompoundTokens(
+  provider,
+  comptrollerContract,
+  compTokenAbi,
+  excludeAddresses,
+  compTokens,
+) {
+  let compTokenAddresses = await comptrollerContract.getAllMarkets();
+  compTokenAddresses = compTokenAddresses.map((addr) => addr.toLowerCase());
+  compTokenAddresses = compTokenAddresses.filter((addr) => excludeAddresses.indexOf(addr) === -1);
+  compTokenAddresses = compTokenAddresses.filter(
+    (addr) => !Object.keys(compTokens).includes(addr),
+  );
```

Comment:
Same functionality, just using chaining in this suggestion.
```suggestion
  compTokenAddresses = compTokenAddresses
    .map((addr) => addr.toLowerCase())
    .filter((addr) => excludeAddresses.indexOf(addr) === -1)
    .filter((addr) => !Object.keys(compTokens).includes(addr));
```